src/components/date-input/_macro.njk

```diff
@@ -26,7 +26,7 @@
                 "id": params.id + "-day",
                 "type": "number",
                 "name": params.day.name,
-                "classes": (" ons-input--error" if params.error else "") + exclusiveClass | default(""),
+                "classes": (" ons-input--error" if (params.error and params.day.error) or (params.error and not params.day.error and not params.month.error and not params.year.error) else "") + exclusiveClass | default(""),
```

Comment:
Do we need the space before `ons-input--error` class?

Reply:
The space is to differentiate the different classes in the input tag

Reply:
Thought it might be, just wanted to double check :)